pkg/ingester/shipper.go

```diff
@@ -233,41 +237,68 @@ func (s *Shipper) blockMetasFromOldest() (metas []*metadata.Meta, _ error) {
 	return metas, nil
 }
 
-func readShippedBlocks(dir string) (map[ulid.ULID]struct{}, error) {
+func readShippedBlocks(dir string) (map[ulid.ULID]time.Time, error) {
 	meta, err := readShipperMetaFile(dir)
+	if err != nil {
+		return readThanosShippedBlocks(dir)
+	}
+
+	shippedBlocks := make(map[ulid.ULID]time.Time, len(meta.Shipped))
+	for _, shippedBlock := range meta.Shipped {
+		shippedBlocks[shippedBlock.BlockID] = shippedBlock.ShippedTime
+	}
+	return shippedBlocks, nil
+}
+
+// readThanosShippedBlocks reads the legacy thanos.shipper.json file.
+// TODO Remove in Mimir 2.7.0
+func readThanosShippedBlocks(dir string) (map[ulid.ULID]time.Time, error) {
+	meta, err := readThanosShipperMetaFile(dir)
 	if errors.Is(err, os.ErrNotExist) {
 		// If the meta file doesn't exist it means the shipper hasn't run yet.
-		meta = &shipperMeta{}
+		meta = &thanosShipperMeta{}
 	} else if err != nil {
 		return nil, err
 	}
 
-	// Build a map.
-	shippedBlocks := make(map[ulid.ULID]struct{}, len(meta.Uploaded))
+	shippedBlocks := make(map[ulid.ULID]time.Time, len(meta.Uploaded))
 	for _, blockID := range meta.Uploaded {
-		shippedBlocks[blockID] = struct{}{}
+		shippedBlocks[blockID] = ulid.Time(blockID.Time()).Add(estimatedShippingDuration)
```

Comment:
Shall we set it to `time.Now()` instead? When shipper runs next time, it will use store `now` to `mimir.shipper.json` file anyway (or timestamp of meta.json file), so next call to `readShippedBlocks` will use that value, and we don't need to provide any "estimated" ship time.